src/explore-education-statistics-common/src/components/form/FormColourInput.tsx

```diff
@@ -4,12 +4,25 @@ import BaseFormInput, {
 import React from 'react';
 
 export interface FormColourInputProps extends FormBaseInputProps {
-  list: string;
+  colours?: string[];
   value?: string;
 }
 
-const FormColourInput = (props: FormColourInputProps) => {
-  return <BaseFormInput {...props} type="color" />;
+const FormColourInput = ({ colours, id, ...props }: FormColourInputProps) => {
+  return (
+    <>
+      <BaseFormInput {...props} id={id} type="color" list={`${id}-colours`} />
+
+      {colours && (
+        <datalist id={`${id}-colours`}>
+          {colours.map((colour, index) => (
+            // eslint-disable-next-line react/no-array-index-key
+            <option key={index} value={colour} />
```

Comment:
If the colour names are unique, could they be used as a key so as to avoid the warning disabling?

Reply:
Nope, there's no guarantee they'll be unique so index is the best we can do